test/support/protocol_utils.rb

```diff
@@ -203,6 +203,66 @@ def assert_reattach
       end
     end
 
+    def assert_locals_result expected, frame_idx: 0
+      case ENV['RUBY_DEBUG_TEST_UI']
+      when 'vscode'
+        # get frameId
+        send_request 'stackTrace',
+                      threadId: 1,
+                      startFrame: 0,
+                      levels: 20
+        res = find_crt_dap_response
+        f_id = res.dig(:body, :stackFrames, frame_idx, :id)
+
+        # get variablesReference
+        send_request 'scopes', frameId: f_id
+        res = find_crt_dap_response
+        locals_scope = res.dig(:body, :scopes).find { |d| d[:presentationHint] == "locals" }
+        locals_reference = locals_scope[:variablesReference]
+
+        # get variables
+        send_request 'variables', variablesReference: locals_reference
+        res = find_crt_dap_response
+
+        expected.each do |exp|
+          if exp[:type] == "String"
+            exp[:value] = exp[:value].inspect
+          end
+        end
+
+        actual_locals = res.dig(:body, :variables).map { |loc| { name: loc[:name], value: loc[:value], type: loc[:type] } }
+      when 'chrome'
+        current_frame = @crt_frames.first
+        locals_scope = current_frame[:scopeChain].find { |f| f[:type] == "local" }
+        object_id = locals_scope.dig(:object, :objectId)
+        send_request "Runtime.getProperties", objectId: object_id
+        res = find_crt_cdp_response
+
+        actual_locals = res.dig(:result, :result).map do |loc|
+          type = loc.dig(:value, :className) || loc.dig(:value, :type).capitalize # TODO: sync this with get_ruby_type
+
+          { name: loc[:name], value: loc.dig(:value, :description), type: type }
+        end
+      end
+
+      failure_msg = FailureMessage.new{create_protocol_message "result:\n#{JSON.pretty_generate res}"}
+
+      actual_locals = actual_locals.sort_by { |h| h[:name] }
+      expected = expected.sort_by { |h| h[:name] }
+
+      expected.each_with_index do |expect, index|
+        actual = actual_locals[index]
+        assert_equal(expect[:name], actual[:name], failure_msg)
+        assert_equal(expect[:type], actual[:type], failure_msg)
+
+        if expect[:value].is_a?(Regexp)
```

Comment:
Do you have a reason why we need to check if the value is Regexp or not?
assert_match can check Regexp and String, right?

Reply:
Because if it's given a string, I expect that the values to be equivalent (`assert_equal`)? I don't want `value: "f"` to match `"foobarbaz"`.

Reply:
Oh, you're right. I misunderstand it. I apologize for it.